dkg/src/dkg_manager/mod.rs

```diff
@@ -117,60 +113,128 @@ impl<DKG: DKGTrait> DKGManager<DKG> {
                 ..
             } = session_state;
-            self.setup_deal_broadcast(start_time_us, &metadata)
-                .await
-                .expect("setup_deal_broadcast() should be infallible");
-        }
 
-        let (agg_trx_tx, mut agg_trx_rx) = aptos_channel::new(QueueStyle::KLAST, 1, None);
-        self.agg_trx_tx = Some(agg_trx_tx);
+            if metadata.dealer_epoch == self.epoch_state.epoch {
+                info!(
+                    epoch = self.epoch_state.epoch,
+                    "Found unfinished and current DKG session. Continuing it."
+                );
+                self.setup_deal_broadcast(start_time_us, &metadata)
+                    .await
+                    .expect("[DKG] setup_deal_broadcast() should be infallible");
+            } else {
+                info!(
+                    cur_epoch = self.epoch_state.epoch,
+                    dealer_epoch = metadata.dealer_epoch,
+                    "Found unfinished but stale DKG session. Ignoring it."
+                );
+            }
+        }
 
-        let mut dkg_start_event_rx = dkg_start_event_rx.into_stream();
         let mut close_rx = close_rx.into_stream();
         while !self.stopped {
             let handling_result = tokio::select! {
                 dkg_start_event = dkg_start_event_rx.select_next_some() => {
-                    self.process_dkg_start_event(dkg_start_event.ok()).await
+                    self.process_dkg_start_event(dkg_start_event)
+                        .await
+                        .map_err(|e|anyhow!("[DKG] process_dkg_start_event failed: {e}"))
                 },
                 (_sender, msg) = rpc_msg_rx.select_next_some() => {
-                    self.process_peer_rpc_msg(msg).await
+                    self.process_peer_rpc_msg(msg)
+                        .await
+                        .map_err(|e|anyhow!("[DKG] process_peer_rpc_msg failed: {e}"))
                 },
-                agg_node = agg_trx_rx.select_next_some() => {
-                    self.process_aggregated_transcript(agg_node).await
+                agg_transcript = agg_trx_rx.select_next_some() => {
+                    self.process_aggregated_transcript(agg_transcript)
+                        .await
+                        .map_err(|e|anyhow!("[DKG] process_aggregated_transcript failed: {e}"))
+
                 },
                 dkg_txn = self.pull_notification_rx.select_next_some() => {
-                    self.process_dkg_txn_pulled_notification(dkg_txn).await
+                    self.process_dkg_txn_pulled_notification(dkg_txn)
+                        .await
+                        .map_err(|e|anyhow!("[DKG] process_dkg_txn_pulled_notification failed: {e}"))
                 },
                 close_req = close_rx.select_next_some() => {
                     self.process_close_cmd(close_req.ok())
-                }
+                },
+                _ = interval.tick().fuse() => {
```

Comment:
Not sure if this is the best practice for logging.. @zekun000 thoughts?

Reply:
it's okay but this will print out the whole transcript bytes?

Reply:
ah i see there's a Debug impl for the transcript